tikv/tikv
# pitr: support modifying the config tikv.log-backup.max-flush-interval online.

### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/14433

What's Changed:
As title~

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 


### Manual test
```
MySQL [(none)]> show config where type='tikv' and name='log-backup.max-flush-interval';
+------+-----------------+-------------------------------+-------+
| Type | Instance        | Name                          | Value |
+------+-----------------+-------------------------------+-------+
| tikv | 10.2.5.31:20162 | log-backup.max-flush-interval | 3m    |
| tikv | 10.2.5.11:20160 | log-backup.max-flush-interval | 3m    |
| tikv | 10.2.5.18:20161 | log-backup.max-flush-interval | 3m    |
+------+-----------------+-------------------------------+-------+


MySQL [(none)]> set config tikv `log-backup.max-flush-interval`="2m";
MySQL [(none)]> show config where type='tikv' and name='log-backup.max-flush-interval';
+------+-----------------+-------------------------------+-------+
| Type | Instance        | Name                          | Value |
+------+-----------------+-------------------------------+-------+
| tikv | 10.2.5.31:20162 | log-backup.max-flush-interval | 2m    |
| tikv | 10.2.5.11:20160 | log-backup.max-flush-interval | 2m    |
| tikv | 10.2.5.18:20161 | log-backup.max-flush-interval | 2m    |
+------+-----------------+-------------------------------+-------+

MySQL [(none)]> set config tikv `log-backup.max-flush-interval`="1m";
MySQL [(none)]> show config where type='tikv' and name='log-backup.max-flush-interval';
+------+-----------------+-------------------------------+-------+
| Type | Instance        | Name                          | Value |
+------+-----------------+-------------------------------+-------+
| tikv | 10.2.5.31:20162 | log-backup.max-flush-interval | 1m    |
| tikv | 10.2.5.11:20160 | log-backup.max-flush-interval | 1m    |
| tikv | 10.2.5.18:20161 | log-backup.max-flush-interval | 1m    |
+------+-----------------+-------------------------------+-------+

MySQL [(none)]> set config tikv `log-backup.max-flush-interval`="0m";
Query OK, 0 rows affected, 3 warnings (0.02 sec)

MySQL [(none)]> show warnings;
+---------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                                                                                                 |
+---------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------+
| Warning | 1105 | bad request to http://10.2.5.31:20182/config: failed to update, error: "the max_flush_interval is too small, it is 0s, and should be greater than 10s." |
| Warning | 1105 | bad request to http://10.2.5.11:20180/config: failed to update, error: "the max_flush_interval is too small, it is 0s, and should be greater than 10s." |
| Warning | 1105 | bad request to http://10.2.5.18:20181/config: failed to update, error: "the max_flush_interval is too small, it is 0s, and should be greater than 10s." |
+---------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------+

```

Find message from tikv.log
```
[2023/03/20 19:21:33.708 +08:00] [INFO] [config.rs:29] [" log backup config changed"] [change="{\"max_flush_interval\": 120000ms}"]
[2023/03/20 19:29:02.639 +08:00] [INFO] [config.rs:29] [" log backup config changed"] [change="{\"max_flush_interval\": 60000ms}"]
```


Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
PITR supports modifying the config tikv.log-backup.max-flush-interval online.
```
